### ISSUE TITLE
Finetuning replication

### ISSUE THREAD

**Issue by yahyunee** [wjq-learning/CBraMod#4]
First of all, thank you for your outstanding work—it has been incredibly insightful.

I am currently attempting to replicate your fine-tuning results, but I have encountered some difficulties in achieving the expected outcomes. Below are the steps I followed:

1. Downloaded and preprocessed the FACED and SEED-V datasets using your provided code.
- I made only minimal adjustments, such as setting the relevant paths.
2. Downloaded pretrained_weights.pth from the Hugging Face repository specified in the README.md file.
3. Ran finetune_main.py with the same parameters as in your setup. The command I used is as follows(for FACED):

`python /CBraMod/finetune_main.py \
    --seed 42 \
    --cuda 0 \
    --epochs 50 \
    --batch_size 64 \
    --lr 1e-4 \
    --weight_decay 5e-2 \
    --optimizer AdamW \
    --clip_value 1 \
    --dropout 0.1 \
    --downstream_dataset FACED \
    --datasets_dir /PREPROCESSED_CBRAMOD_DATA/FACED\
    --num_of_classes 9 \
    --model_dir /CBraMod/data/lucy/models_weights/Big/BigFaced \
    --num_workers 16 \
    --label_smoothing 0.1 \
    --frozen False \
    --use_pretrained_weights True \
    --foundation_dir pretrained_weights/pretrained_weights.pth`

I got result as below accuracy of 0.23 for SEED-V and 0.25 for FACED which is far below the reported accuracy.
![Image](https://github.com/user-attachments/assets/29b89770-5c11-48ba-a1e5-dd32cdfc4025)

I've checked number of samples for each dataset and it was identical to the number of samples reported in the paper.
I would greatly appreciate any guidance or suggestions on what I might be missing. Thank you in advance for your time and assistance!

**Comment by wjq-learning** [comment]
Hello, thank you for your detailed explanation of the steps you followed. To better assist you in identifying the issue, could you please provide the log files generated during the training process? The logs would help us understand the training dynamics, such as loss values, accuracy trends, and potential warnings or errors that might have occurred.

**Comment by wjq-learning** [comment]
Hi，we find the hyperparameter frozen is set to True in the log. Please set it to False.

**Comment by yahyunee** [comment]
Thank you for finding out my mistake! I figured out there was a problem in the parser of `finetune_main.py` as it get arguments `--frozen` and `--use_pretrained_weights` **as string not boolean type**. I've adjusted the code using function that sets argument type as boolean, which was addressed below.
https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse

Should I make a pull request?
+ I got the accuracy that matches the one reported in FACED finetuning task!! Thank you for your support
